s3backgrounddelete/s3backgrounddelete/object_recovery_msgbus.py

```diff
@@ -0,0 +1,192 @@
+
+    def close(self):
+        """Closure and cleanup for ObjectRecoveryMsgbus."""
+        pass
+
+    def __process_msg(self, msg):
+        """Loads the json message and sends it to validation and processing."""
+        self._logger.info(
+            "Processing following records in consumer: " + msg)
+        try:
+            # msg: {"Key": "egZPBQAAAAA=-ZQIAAAAAJKc=",
+            #       "Value": "{\\"index_id\\":\\"egZPBQAAAHg=-YwIAAAAAJKc=\\",
+            #       \\"object_layout_id\\":1,
+            #        \\"object_metadata_path\\":\\"object1\\"}\\n"}
+            probable_delete_records = json.loads(msg)
+
+            if probable_delete_records:
+                validator = ObjectRecoveryValidator(
+                    self._config, probable_delete_records, self._logger)
+                validator.process_results()
+
+        except (KeyError, ValueError) as ex:    # Bad formatted message. Will discard it
+            self._logger.error("Failed to parse JSON data due to: " + str(ex))
+            return True
+        except Exception as ex:
+            self._logger.error(str(ex))
+            return False
+        return True
+
+    def __loadmsgbuslibrary(self):
+        """Load message bus library."""
+        #Basically loads msgbus lib and reads its config file
+        self._logger.debug("Instantiating S3MessageBus")
+        self.__msgbuslib = S3CortxMsgBus()
+        self._logger.debug("Connecting to S3MessageBus")
+        ret,msg = self.__msgbuslib.connect()
+        if not ret:
+            self._logger.error("msgbus connect failed {}".format(msg))
+            self.__msgbuslib = None
+        return ret
+
+    def receive_data(self,
+        consumer_id = None,
+        consumer_group = None,
+        msg_topic = None,
+        autoack = None,
+        offset = None):
+        """Initializes consumer, connects and receives messages from message bus."""
+        try:
+            if not self.__msgbuslib:
+                ret = self.__loadmsgbuslibrary()
+                if not ret:
+                    time.sleep(self._sleep_time)
+                    self.receive_data()
+
+            #Over here we will have msgbuslib loaded
+            if not consumer_id:
+                consumer_id = self._config.get_msgbus_consumer_id_prefix() + str(socket.gethostname())
+            
+            if not consumer_group:
+                consumer_group = self._config.get_msgbus_consumer_group()
+            
+            if not msg_topic:
+                msg_topic = self._config.get_msgbus_topic()
+
+            if not autoack:
+                autoack = False
+
+            if not offset:
+                offset = 'earliest'
+
+            self._logger.debug("Setting up S3MessageBus for consumer")
+            ret, msg = self.__msgbuslib.setup_receive(consumer_id,
+                consumer_group, msg_topic, autoack, offset)
+            if ret:
+                while True:
```

Comment:
Use 3 explicit while loops

Even if you are recursing, pass the parameters received to receive_data()

Reply:
seperated the setup part so only 2 while loops are required.